File: apps/rn/src/domain/janela.ts
import { horaSaoPaulo } from "./budgetCalculator";

export function janelaNaHora(hora: number): string {
  if (hora >= 5 && hora <= 9) return "cafe";
  if (hora >= 10 && hora <= 11) return "lanche_manha";
  if (hora >= 12 && hora <= 14) return "almoco";
  if (hora >= 15 && hora <= 17) return "lanche_tarde";
  if (hora >= 18 && hora <= 21) return "janta";
  return "ceia";
}

export function janelaAgora(instant: Date = new Date()): string {
  return janelaNaHora(horaSaoPaulo(instant));
}

export function tituloJanela(id: string): string {
  switch (id) {
    case "cafe":
      return "Café";
    case "lanche_manha":
      return "Lanche manhã";
    case "almoco":
      return "Almoço";
    case "lanche_tarde":
      return "Lanche tarde";
    case "janta":
      return "Janta";
    default:
      return "Ceia";
  }
}

export const JANELAS = ["cafe", "lanche_manha", "almoco", "lanche_tarde", "janta", "ceia"] as const;

File: apps/android/app/src/main/java/com/nutri/android/domain/Janela.kt
package com.nutri.android.domain

fun janelaNaHora(hora: Int): String = when (hora) {
    in 5..9 -> "cafe"
    in 10..11 -> "lanche_manha"
    in 12..14 -> "almoco"
    in 15..17 -> "lanche_tarde"
    in 18..21 -> "janta"
    else -> "ceia"
}

fun tituloJanela(id: String): String = when (id) {
    "cafe" -> "Café"
    "lanche_manha" -> "Lanche manhã"
    "almoco" -> "Almoço"
    "lanche_tarde" -> "Lanche tarde"
    "janta" -> "Janta"
    else -> "Ceia"
}
